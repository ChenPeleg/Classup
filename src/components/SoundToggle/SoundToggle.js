import React from "react";
import volumeOn from "./volumeon.png";
import volumeOff from "./volumeoff.png";
import styles from "./SoundToggle.module.scss";
const SoundToggle = (props) => {
  return (
    <img
      src={props.soundOn ? volumeOn : volumeOff}
      className={styles.soundToggle}
      onClick={() => {
        props.soundHandler();
      }}
      alt="volume toggle"
    />
  );
};
export default SoundToggle;
