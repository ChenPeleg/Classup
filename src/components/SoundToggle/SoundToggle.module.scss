.soundToggle {
  height: 24px * 1.5;
  transition: all 0.5s;
  cursor: pointer;
}

.in_ToolBar {
 
  padding-top: 1%;
  height: 100%; 
  margin: 0px;
  vertical-align: "middle";
  margin: " auto 0 ";
  position: absolute;
  right: 0%;
  
}
.in_SideDrawer {
 position: relative;
 vertical-align: middle;
 
}