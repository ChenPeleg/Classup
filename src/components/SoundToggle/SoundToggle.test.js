import React from "react"
import { shallow } from "enzyme"
import SoundToggle from "./SoundToggle"
const mockClick = jest.fn();
const soundtog = shallow(<SoundToggle soundOn={true} soundHandler={mockClick} />)
 
describe("<Sound Toggle>", () => {
    it('renders correctly', () => {
        expect(soundtog).toMatchSnapshot()
    })
    it("fires sound toggle event", () => {
        soundtog.simulate('click')
        expect(mockClick).toHaveBeenCalled()
    });
    it("contains image", () => {
        expect(soundtog.contains('img')).toEqual(false);
    })
})