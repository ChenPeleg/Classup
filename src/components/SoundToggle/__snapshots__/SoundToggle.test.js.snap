// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sound Toggle> renders correctly 1`] = `
<img
  alt="volume toggle"
  className="soundToggle in_ToolBar"
  onClick={[Function]}
  src="volumeon.png"
/>
`;
